text/0000-generalized-type-ascription.md

```diff
@@ -540,7 +379,7 @@ you can resolve the errors by writing:
 fn main() {
     println!("{:?}", (0..10).map(|x| x % 3 == 0).collect() : Vec<_>);
 
-    let _ = ("1234".parse().unwrap() : usize).into() : Box<_>;
+    let _ = "1234".parse().unwrap(): usize.into(): Box<_>;
```

Comment:
Why drop these parentheses? This seems far less readable.

Reply:
Ah, thanks. The latter sounds good to me: it's *possible* to write it that way, but I think we should encourage parentheses whenever it's on one line.

Reply:
That might be sensible; we might want to gain some experience first / I'll leave the one-line issue to the style team... :)

Reply:
I can tell you that I *did* parse it as `("1234".parse().unwrap()): (usize.into())` at first, so I'd strongly discourage this style. I understand that it's reasonable when it's broken into multiple lines, but when there's no whitespace in between at all, it looks very confusing, and after you've resolved the ambiguity mentally, it "only" looks wrong.

Reply:
@H2CO3 Yeah sure, you are probably right. All I'm saying is that rustfmt style decisions can be postponed to some other time. For now, I think it suffices to know that there is a sensible styling for one-liner chains + ascription.

Reply:
@Centril of course! I'm just saying that it will probably not be a good idea to *promote* the style where the spacing suggests something different from the semantics. I agree that it should work, though (this argument is true with most whitespace-related styling issues in general).

Reply:
```rust
     let _ = "1234".parse().unwrap():usize.into(): Box<_>; 
```

I'd suggest at least remove the space before `usize`. It could still be read as `("1234".parse().unwrap()):(usize.into())` but not as strongly as the current format.

Reply:
What about a space after the type?

```rust
     let _ = "1234".parse().unwrap():usize .into(): Box<_>; 
```

Reply:
When y'all figure out the bikeshed / a phrasing that you like, please send a PR / GH suggestion my way... 😂

Reply:
I am in favor or removing the spaces.

At least for my brain spaces have a higher grouping precedence than dots. In other words when I read a chain then I try to parse the dots as separators, so `.unwrap():usize` is a single cluster within the sentence saying *call unwrap on whatever is on the left hand and require the return type to be usize* .
But if you mix spaces and dots then `usize.into()` becomes a separate cluster since spaces separate more strongly than dots. The resulting subunit is nonsensical at first glance because it has been separated from its colon that indicates it's a type ascription.